zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -2499,7 +2503,7 @@ public Object call() throws Exception {
    * @throws IllegalStateException
    *           if the connection timed out due to thread interruption
    */
-  public void connect(final long maxMsToWaitUntilConnected, Watcher watcher)
+  public void connect(final long maxMsToWaitUntilConnected, @Nullable Watcher watcher)
```

Comment:
curious would there be a case when meta-client or caller explicitly passes "null" as watcher? (meaning it doesn't want to use native zk client as watcher but just null)

Reply:
That basically means there is no watcher for zookeeper notification, and even NPE, I don't think it's a valid use case. Let me know if that makes sense to you.

Reply:
Umm if it's not expected then it's fine. I was just thinking that it should not be silently ignored because of this Nullable.